### PR TITLE
Switch to new localization pipeline

### PR DESCRIPTION
**Pull request by Colengms**
Addresses: https://github.com/microsoft/vscode-cpptools/issues/7433

The import script now pulls translations from the Loc repo instead of a file share.  Multi-checkout was not working for me, possible due to spanning GitHub and DevOps.  So, it clones the loc repo using a PAT.

The import script was also updated to work properly when deployed via YML.

The export script now uses the OneLocBuild DevOps.  (I think I will also need to point them to the build pipeline that uses it, after this is checked in, as it just generates artifacts that I think they will need to pick up).

Removed a "\n" from a description string, as somewhere in the pipeline a "\r" is also being added to that.  No other descriptions add new lines, so removed that instance.

### CODE REVIEW

File: Extension/translations_auto_pr.js [hunk 1]
```diff
@@ -66,22 +70,16 @@ function sleep(ms) {
...
 directories.forEach(folderName => {
-    let sourcePath = `${rootSourcePath}\\${folderName}\\Projects\\Src\\vscode-extensions\\vscode-${locProjectName}.${folderName}.xlf`;
+    let sourcePath = `${rootSourcePath}\\${folderName}\\vscode-cpptools.xlf`;
     let destinationPath = `../vscode-translations-import/${folderName}/vscode-extensions/vscode-${locProjectName}.xlf`;
-    console.log(`Copying "${sourcePath}$" to "${destinationPath}"`);
+    console.log(`Copying "${sourcePath}" to "${destinationPath}"`);
     fs.copySync(sourcePath, destinationPath);
 });
 
+
```
Comment (sean-mcmanus) [inline Extension/translations_auto_pr.js]: Is this extra newline intentional?

File: Extension/translations_auto_pr.js [hunk 2]
```diff
@@ -115,14 +113,21 @@ if (!hasAnyChanges()) {
     return;
 }
 
-// Commit changes files.
-console.log(`Commiting changes (git commit -m "${commitComment}")`);
-cp.execSync(`git commit -m "${commitComment}"`);
+// Set up user and permissions(Never run this locally)
```
Comment (sean-mcmanus) [inline Extension/translations_auto_pr.js]: What happens if it's run locally? Should there be some guard against that in case?

**Review by sean-mcmanus — APPROVED** [review]

**Comment by michelleangela** [comment]
Is there public documentation for how the "new localization pipeline" works that we can save/reference in our internal docs for future understanding or maintenance?

File: Extension/translations_auto_pr.js [hunk 3]
```diff
@@ -15,9 +15,12 @@ let repoName = process.argv[3];
 let locProjectName = process.argv[4];
 let authUser = process.argv[5];
 let authToken = process.argv[6];
+let userFullName = process.argv[7];
+let userEmail = process.argv[8];
+let locRepoPath = process.argv[9];
 
 if (!repoOwner || !repoName || !locProjectName || !authUser || !authToken) {
```
Comment (michelleangela) [inline Extension/translations_auto_pr.js]: Also add check for the new vars `user_full_name` and `user_email` and `loc_repo_path`? Or add comment that states which args are required at minimum to be correct?

File: Extension/translations_auto_pr.js (hunk 2 above)
Comment (Colengms) [reply Extension/translations_auto_pr.js]: I set the values using script args, so if run locally the right local values could be passed in.  I had to set them as there is an error if unset on commit, on the build vm.  I'm not sure how to guard against it.  I could do so more digging to find a way.

**Comment by Colengms** [comment]
> Is there public documentation for how the "new localization pipeline" works that we can save/reference in our internal docs for future understanding or maintenance?

Unfortunately, no, not really.  I'm not getting a lot of support from the loc team, so I've actually been struggling to ensure we can continue to keep this process automated.  There is this: https://dev.azure.com/ceapex/CEINTL/_git/SoftwareLocalization?path=%2Fsrc%2FOneLocBuild  (Which Jonathan referred me to, and the Loc team did not).  Which we're able to leverage to generate files for submitting into Juno, but we can't use its functionality to generate PRs, as that only works for DevOps repos and would not do the additional step we need of actually ingesting the strings into the i18n directory.  I did get some 'instructions' from the loc team, which arrived after I had composed scripts to do the same things.

File: Build/loc/TranslationsImport.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    workingDirectory: '$(Build.SourcesDirectory)'
+
+- task: CmdLine@2
+  inputs:
+    script: 'cd Extension && yarn install'
+    workingDirectory: '$(Build.SourcesDirectory)'
+
+- task: CmdLine@2
+  inputs:
+    script: 'cd Extension && node ./translations_auto_pr.js microsoft vscode-cpptools cpptools csigs $(csigsPat) "cpptools service account" cpptools@microsoft.com $(Build.SourcesDirectory)/../VSCodeExt'
```
Comment (bobbrow) [inline Build/loc/TranslationsImport.yml]: Can you remove the direct reference to the internal account as well?  We are retiring this account anyway.

File: Build/loc/TranslationsExport.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# ==================================================================================
+# Pipeline for VsCodeExtension-Localization build definition
+# Runs OneLocBuild task to localize xlf file
+#
+# Triggered only by daily schedule
```
Comment (sean-mcmanus) [inline Build/loc/TranslationsExport.yml]: This comment seems no longer valid.

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]

File: Build/loc/TranslationsImport.yml (hunk 4 above)
Comment (Colengms) [reply Build/loc/TranslationsImport.yml]: I need to set the email and user name of the git commit to something.  That something will be displayed in the (public) git log, so there is no use in hiding it here.  Do you have a suggestion on what that user name and email should be?

**Review by sean-mcmanus — APPROVED** [review]

File: Build/loc/TranslationsImport.yml (hunk 4 above)
Comment (Colengms) [reply Build/loc/TranslationsImport.yml]: Previously, we were picking up the default configured git user name and email from the build machine.  i.e.:
DevDiv Build Lab - Dev14 <dlab14@microsoft.com>

But, when using YML, the build machine is not configured with a user name or email, and cannot commit unless I set them.

Provided that the PR is squashed when committed, that info will be replaced with the name and email of associated with the PR (csigs) :

csigs <csigs@users.noreply.github.com>

I suppose I could use that same account information for the commit.

**Review by michelleangela — APPROVED** [review]

**Review by bobbrow — APPROVED** [review]

**Pull request merged**